src/enums.h

```diff
@@ -714,4 +714,42 @@ enum DamageAnalyzerImpactType
 	RECEIVED = 2
 };
 
+enum class LootCategory : uint8_t
+{
+	NONE = 0,
+	ARMOR = 1,
+	AMULET = 2,
+	BOOTS = 3,
+	CONTAINER = 4,
+	DECORATION = 5,
+	FOOD = 6,
+	HELMET = 7,
+	LEGS = 8,
+	OTHERS = 9,
+	POTIONS = 10,
+	RINGS = 11,
+	RUNES = 12,
+	SHIELDS = 13,
+	TOOLS = 14,
+	VALUABLES = 15,
+	AMMO = 16,
+	AXE = 17,
+	CLUB = 18,
+	DISTANCE = 19,
+	SWORD = 20,
+	WAND = 21,
+	PREMSCROLL = 22,
+	STORECOIN = 23,
+	CREATURE_PRODUCT = 24,
+	QUIVER = 25,
+	UNKNOWN = 26,
+	STASH = 27,
+	UNSORTED = 28,
+	WEAPONS_ALL = 29,
+	GOLD = 30,
+	UNASSIGNED = 31,
+
+	LAST = UNASSIGNED
```

Comment:
let's avoid potential variable name conflict
```suggestion
	LOOT_CATEGORY_NONE = 0,
	LOOT_CATEGORY_ARMOR = 1,
	LOOT_CATEGORY_AMULET = 2,
	LOOT_CATEGORY_BOOTS = 3,
	LOOT_CATEGORY_CONTAINER = 4,
	LOOT_CATEGORY_DECORATION = 5,
	LOOT_CATEGORY_FOOD = 6,
	LOOT_CATEGORY_HELMET = 7,
	LOOT_CATEGORY_LEGS = 8,
	LOOT_CATEGORY_OTHERS = 9,
	LOOT_CATEGORY_POTIONS = 10,
	LOOT_CATEGORY_RINGS = 11,
	LOOT_CATEGORY_RUNES = 12,
	LOOT_CATEGORY_SHIELDS = 13,
	LOOT_CATEGORY_TOOLS = 14,
	LOOT_CATEGORY_VALUABLES = 15,
	LOOT_CATEGORY_AMMO = 16,
	LOOT_CATEGORY_AXE = 17,
	LOOT_CATEGORY_CLUB = 18,
	LOOT_CATEGORY_DISTANCE = 19,
	LOOT_CATEGORY_SWORD = 20,
	LOOT_CATEGORY_WAND = 21,
	LOOT_CATEGORY_PREMIUM_SCROLL = 22,
	LOOT_CATEGORY_STORE_COIN = 23,
	LOOT_CATEGORY_CREATURE_PRODUCT = 24,
	LOOT_CATEGORY_QUIVER = 25,
	LOOT_CATEGORY_UNKNOWN = 26,
	LOOT_CATEGORY_STASH = 27,
	LOOT_CATEGORY_UNSORTED = 28,
	LOOT_CATEGORY_WEAPONS_ALL = 29,
	LOOT_CATEGORY_GOLD = 30,
	LOOT_CATEGORY_UNASSIGNED = 31,

	LOOT_CATEGORY_LAST = LOOT_CATEGORY_UNASSIGNED
```

Reply:
enum class does exactly that, avoids name conflicts, in the second part of the PR I'll add a function registerEnumClass to register these variables in lua as `LOOT_CATEGORY_...`